apache/incubator-sedona
# [SEDONA-3] First release: Set up POM for uploading snapshots and releases

## Is this PR related to a proposed Issue?

https://issues.apache.org/jira/browse/SEDONA-3

## What changes were proposed in this PR?

1. Remove JTS Git submodule, use JTS 1.17.1 official release with 5 additional JTS index files copied from JTS 1.18 (see my PR: https://github.com/locationtech/jts/pull/634)
2. Update the POM file accordingly
3. Move scalastyle_config.xml to the root just like what Spark does

## How was this patch tested?

Successfully published SNAPSHOTs to Nexus: https://repository.apache.org/ You can verify this by searching sedona in the search box

The following commands are used to cut the snapshots: https://gist.github.com/jiayuasu/849e1f3bf7a2dd11593ca27c14e9e92d

```
Maven:

https://infra.apache.org/publishing-maven-artifacts.html
https://repository.apache.org/

- Publish Sedona for Spark 3.0 and Scala 2.12
python3 spark-version-converter.py spark3
mvn clean -Darguments="-DskipTests" release:prepare -DdryRun=true -DautoVersionSubmodules=true -Dresume=false
- Publish a snapshot
mvn deploy -DskipTests

- Publish Sedona for Spark 2.4 and Scala 2.11
python3 spark-version-converter.py spark2
mvn clean release:prepare -DdryRun=true -DautoVersionSubmodules=true -Dresume=false -DcheckModificationExcludeList=sql/src/main/scala/org/apache/sedona/sql/UDF/UdfRegistrator.scala,sql/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/JoinQueryDetector.scala,sql/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/TraitJoinQueryExec.scala -Darguments="-DskipTests -Dscala=2.11 -Dspark=2.4"
- Publish a snapshot
mvn deploy -DskipTests -Dscala=2.11 -Dspark=2.4

- Publish Sedona for Spark 2.4 and Scala 2.12
python3 spark-version-converter.py spark2
mvn clean release:prepare -DdryRun=true -DautoVersionSubmodules=true -Dresume=false -DcheckModificationExcludeList=sql/src/main/scala/org/apache/sedona/sql/UDF/UdfRegistrator.scala,sql/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/JoinQueryDetector.scala,sql/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/TraitJoinQueryExec.scala -Darguments="-DskipTests -Dscala=2.12 -Dspark=2.4"
- Publish a snapshot
mvn deploy -DskipTests -Dscala=2.12 -Dspark=2.4
```

## Did this PR include necessary documentation updates?
